### ISSUE TITLE
About training SSD with different annotation format

### ISSUE THREAD

**Issue by lxd7714059** [weiliu89/caffe#59]
Can i train my own data with annotation format .txt？

**Comment by weiliu89** [comment]
Yes. It supports annotation in .txt format. Check out details at [here](https://github.com/weiliu89/caffe/blob/ssd/src/caffe/util/io.cpp#L461).

**Comment by lxd7714059** [comment]
Thank you very much!

My .txt is like this: class_id, xmin, ymin, xmax, ymax
When i create lmdb file using create_data.sh, the error is "does not exist: /home/lxd/data/GSTDB/Annotations/00000.txt". 
But i do have the file in that directory.
My create_data.sh is like this:

cur_dir=$(cd $( dirname ${BASH_SOURCE[0]} ) && pwd )
root_dir=$cur_dir/../..

cd $root_dir

redo=1
data_root_dir="$HOME/data"
dataset_name="GSTDB"
mapfile="$root_dir/data/$dataset_name/labelmap_voc.prototxt"
anno_type="detection"
db="lmdb"
min_dim=0
max_dim=0
width=0
height=0
label_type="txt"

extra_cmd="--encode-type=ppm --encoded"
if [ $redo ]
then
  extra_cmd="$extra_cmd --redo"
fi
for subset in test trainval
do
  python $root_dir/scripts/create_annoset.py --anno-type=$anno_type --label-type=$label_type --label-map-file=$mapfile --min-dim=$min_dim --max-dim=$max_dim --resize-width=$width --resize-height=$height --check-label $extra_cmd $data_root_dir $root_dir/data/$dataset_name/$subset.txt $data_root_dir/$dataset_name/$db/$subset"_"$db examples/$dataset_name
done

Can you tell me what's wrong with that?

**Comment by weiliu89** [comment]
I am not sure. You can look into scripts/create_annoset.py and see what is the error.

**Comment by conner99** [comment]
I get the same error on Linux.

When i comment out [sys.exit()](https://github.com/weiliu89/caffe/blob/ssd/scripts/create_annoset.py#L96) i see the error, but the LMDB files get generated and the output [print cmd](https://github.com/weiliu89/caffe/blob/ssd/scripts/create_annoset.py#L158) looks to be correct.

I don't know if these files are correct.
So i use _caffe/build/tools/convert_annoset_ directly.

**Comment by ShihJenYu** [comment]
check your txt  if you created txt on windows , you can try it (control character \r\n replace to \n ).
